ast/desugar/Desugar.cc

```diff
@@ -1677,9 +1677,9 @@ TreePtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what) {
                 auto res = unsupportedNode(dctx, redo);
                 result = std::move(res);
             },
-            [&](parser::EncodingLiteral *redo) {
-                auto res = unsupportedNode(dctx, redo);
-                result = std::move(res);
+            [&](parser::EncodingLiteral *encodingLiteral) {
```

Comment:
Looks like the behavior of Ruby is that the value of `__ENCODING__` is figured out by the parser. e.g., there's some state in the parser that looks for an `# encoding: ...` comment, so that when the `__ENCODING__` node type is encountered, it can look up an existing `#<Encoding:UTF-8>` or whatever instance. You can see that this is happening by looking at what the Ruby VM compiles `__ENCODING__` to:

```
❯ ruby --dump=insns -e '__ENCODING__'
== disasm: #<ISeq:<main>@-e:1 (1,0)-(1,12)> (catch: FALSE)
0000 putobject                    #<Encoding:UTF-8>                   (   1)[Li]
0002 leave
```

e.g., before it's even started to run the file, Ruby's bytecode compiler has already found a hard-coded global `Encoding` instance, and emitted bytecode instructions to push that value onto the stack.

No action required here, I just wanted to record something I learned about how `__ENCODING__` works.